airflow/utils/dag_processing.py

```diff
@@ -141,7 +141,7 @@ def waitable_handle(self):
 class DagParsingStat(NamedTuple):
     """Information on processing progress"""
 
-    file_paths: List[str]
+    num_file_paths: int
```

Comment:
let's further challenge this part: why do we need `num_file_paths` or earlier `file_paths`?

Where we really need this `DagParsingStat` is `self._sync_metadata`, in which we don't use `num_file_paths` or earlier `file_paths`

Reply:
Checking `self._sync_metadata` also leads me to something else:

https://github.com/apache/airflow/blob/1bec3b21266764f367431ab5d9a5b75f52b9b6d2/airflow/utils/dag_processing.py#L323-L326 is doing exactly the same thing in `self._process_message()`, which has been invoked just before L323-326. 

A clean-up refactor to do? Or I misunderstood something?

Reply:
Fair point yes, this is only used for tests and could possibly be removed. Let me take a look.

Reply:
Yes, done, was only used in one test, and that was strictly speaking not testing the right thing anymore, just a likely side-effect.

```diff
-        assert result.num_file_paths == 3
+        assert sum(stat.run_count for stat in manager._file_stats.values()) == 3
```

See 4d8da57eb for full diff.

Good call @XD-DENG

Reply:
> Checking `self._sync_metadata` also leads me to something else:
> 
> https://github.com/apache/airflow/blob/1bec3b21266764f367431ab5d9a5b75f52b9b6d2/airflow/utils/dag_processing.py#L323-L326
> 
> is doing exactly the same thing in `self._process_message()`, which has been invoked just before L323-326.
> A clean-up refactor to do? Or I misunderstood something?

Thanks @ashb . How about this question?

We can address it separately of course, but want to confirm if I overlooked anything regarding this.

Reply:
Yeah, that looks like a bug. PR comin.

Reply:
lol yeah, good catch @XD-DENG `self. _sync_metadata` is redundant here